extensions/opencv/CaptureRTSPFrame.cpp

```diff
@@ -116,17 +121,28 @@ void CaptureRTSPFrame::onSchedule(core::ProcessContext *context, core::ProcessSe
     rtspURI.append(rtsp_uri_);
   }
 
-  cv::VideoCapture capture(rtspURI.c_str());
-  video_capture_ = capture;
-  video_backend_driver_ = video_capture_.getBackendName();
+  rtsp_url_ = rtspURI;
+
 }
 
 void CaptureRTSPFrame::onTrigger(const std::shared_ptr<core::ProcessContext> &context,
                                  const std::shared_ptr<core::ProcessSession> &session) {
   auto flow_file = session->create();
 
-  cv::Mat frame;
+  try {
+    video_capture_.open(rtsp_url_);
+    video_backend_driver_ = video_capture_.getBackendName();
+  } catch (const cv::Exception &e) {
+    logger_->log_error("Unable to open RTSP stream: %s", e.what());
+    session->transfer(flow_file, Failure);
```

Comment:
This creates an empty FF, so that means if we can't open the URL we could potentially create many flow files that are empty. The typical approach is to yield in the context. You can still create a failure flow file ( though I would suggest exploring if this is the correct route ), but you should probably yield here.